openfisca_core/holders.py

```diff
@@ -446,20 +430,34 @@ def extra_params_to_json_key(extra_params, period):
                 for cell in array.tolist()
                 ]
         value_json = {}
-        if self._array_by_period is not None:
-            for period, array_or_dict in self._array_by_period.iteritems():
-                if type(array_or_dict) == dict:
+        for period, array_or_dict in self._memory_storage._arrays.iteritems():
+            if type(array_or_dict) == dict:
+                value_json[str(period)] = values_dict = {}
```

Comment:
Where is `values_dict`is used ? The method returns `value_json`but I can't find a usage of `values_dict `appart from being assigned values :/

Reply:
This is confusing and I'm rewriting it, but to answer the question:
- `value_json[str(period)] = values_dict = {}` creates a `dict`, stores it in a variable named `values_dict`, and also assign `value_json[str(period)]` to this `dict`.
- Assigning values to `values_dict` is equivalent to assigning values to `value_json[str(period)]`, as these two reference the same dictionary.
- In the end, the returned value `value_json ` has been modified by the assignation

The goal of `values_dict ` is just to be a shortcut towards ` value_json[str(period)]`.

Reply:
Thanks for the explanation :)